src/internal/ButtonBase.js

```diff
@@ -121,6 +103,8 @@ export default class ButtonBase extends Component {
   button = null;
   keyboardFocusTimeout = undefined;
 
+  focus = () => this.button.focus();
```

Comment:
I think that the imperative API is the best one. It's simpler to use to deprecated, to warn when not used correctly, it abstract the underlying dom implementation.